### ISSUE TITLE
Unify CLI arg arg definition and generation

### ISSUE THREAD

**Issue by SaschaCowley** [nvaccess/nvda#16956]
### Detailed description of the issue

NVDA's command line arguments and restart/update logic are currently repeated in multiple places, which violates the DRY principle and reduces NVDA's maintainability. For instance:

- The `argParse.ArgumentParser` used by NVDA is in `nvda.pyw`, and cannot be imported for inspection by any other files;
- Logic about what arguments to pass on to a new or restarted copy of NVDA is contained in `source/updateCheck.py`, `source/core.py`, and possibly other places.
- Unifying these to a set of helper functions that are accessible wherever needed would improve the maintainability of NVDA and reduce some of the repetition currently present in the source.

### Why are the other templates not appropriate in this case?

This not a request for a bug fix or new feature, just a request for the internal code to be cleaned up.

### Have you asked for advice on how to report this issue via a community discussion? If so, please link to the discussion

No.

### Steps to reproduce or illustrate the issue (if applicable)

Look at the code, in particular the files mentioned above. Notice that logic is repeated. Consider how this makes testing more difficult, behaviour less predictable, and updating code potentially more difficult.

### Expected outcome or behaviour (if applicable)

There should be a "single source of truth" for our CLI args and their names. Logic should not be repeated across the code base.

### Screenshots, logs, and other attachments (if applicable)

NA

### System configuration or software environment (if applicable)

NA

### Additional information (if applicable)

NA
